tools/codegen/gofilegen/datasource/data_source_file.go

```diff
@@ -59,3 +63,56 @@ func GeneratePluralGoCode(input *codespec.Resource) ([]byte, error) {
 	}
 	return formattedResult, nil
 }
+
+// pluralizeName converts a resource name to its plural form for data sources.
+// For names ending with "_api", it inserts "s" before "_api".
+// E.g., "stream_instance_api" becomes "stream_instances_api"
+// This approach may not always be correct & should be improved in the future, for example,
+// "resource_policy_api" becomes "resource_policys_api" instead of "resource_policies_api"
+func pluralizeName(name string) string {
+	const apiSuffix = "_api"
+
+	if strings.HasSuffix(name, apiSuffix) {
+		return name[:len(name)-len(apiSuffix)] + "s" + apiSuffix
+	}
+	return name + "s"
+}
+
+// getQueryParams extracts query parameters from plural data source attributes.
+// Query parameters are optional top-level attributes
```

Comment:
```suggestion
// Assumption made is query parameters are optional top-level attributes
```
Would keep this explicit for future reference

Reply:
updated